RxCalculator/Source/ViewController.swift

```diff
@@ -0,0 +1,143 @@
+            rightHandSideTextField.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -47),
+            
+            valueLabel.topAnchor.constraint(equalTo: rightHandSideTextField.bottomAnchor, constant: 20),
+            valueLabel.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: 100),
+            valueLabel.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -100),
+            
+            buttonView.topAnchor.constraint(equalTo: valueLabel.bottomAnchor, constant: 28),
+            buttonView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: 47),
+            buttonView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -47)
+        ])
+        
+        leftHandSideTextField.heightAnchor.constraint(equalToConstant: 35).isActive = true
+        rightHandSideTextField.heightAnchor.constraint(equalToConstant: 35).isActive = true
+        valueLabel.heightAnchor.constraint(equalToConstant: 22).isActive = true
+    }
+    
+    private func bind() {
+        leftHandSideTextField.rx.text.orEmpty
+            .map { Int($0) ?? 0 }
+            .bind(to: viewModel.left)
+            .disposed(by: disposeBag)
+        
+        rightHandSideTextField.rx.text.orEmpty
+            .map { Int($0) ?? 0 }
+            .bind(to: viewModel.right)
+            .disposed(by: disposeBag)
+        
+        buttonView.plusButton.rx.tap
+            .map { Operation.plus }
+            .bind(onNext: { [weak self] op in
+                self?.viewModel.calculate(operation: op)
+            })
+            .disposed(by: disposeBag)
+        
+        buttonView.subtractButton.rx.tap
+            .map { Operation.subtraction }
+            .bind(onNext: { [weak self] op in
+                self?.viewModel.calculate(operation: op)
+            })
+            .disposed(by: disposeBag)
+        
+        buttonView.multiplyButton.rx.tap
+            .map { Operation.multiplication }
+            .bind(onNext: { [weak self] op in
+                self?.viewModel.calculate(operation: op)
+            })
+            .disposed(by: disposeBag)
+        
+        buttonView.divideButton.rx.tap
+            .map { Operation.divide }
+            .bind(onNext: { [weak self] op in
+                self?.viewModel.calculate(operation: op)
+            })
+            .disposed(by: disposeBag)
+        
+        //output
+        viewModel.result
+            .map {  String($0) }
+            .bind(to: valueLabel.rx.text)
+            .disposed(by: disposeBag)
+    }
+}
+    
+extension ViewController {
+    private func checkEmptyInput() {
+        Observable.merge(buttonView.plusButton.rx.tap.map{ $0 }, buttonView.multiplyButton.rx.tap.map{ $0 }, buttonView.subtractButton.rx.tap.map{ $0 }, buttonView.divideButton.rx.tap.map{ $0 })
+            .subscribe(onNext: { [weak self] in
```

Comment:
줄바꿈 해주면 훨씬 보기 편할 것 같아요~
```suggestion
        Observable.merge(
        buttonView.plusButton.rx.tap.map{ $0 }, 
        buttonView.multiplyButton.rx.tap.map{ $0 }, 
        buttonView.subtractButton.rx.tap.map{ $0 }, 
        buttonView.divideButton.rx.tap.map{ $0 }
        )
            .subscribe(onNext: { [weak self] in
```

인덴트가 이상하긴 한데 이런식으로요!

Reply:
그리고 rx.tap을 위에서 ViewModel이랑 binding할때 구독하고 있는 데, 여기 함수에서 또 구독하고 있네요 ? 🤔

Reply:
이거슨 쓰면서도 저도 계속 그 고민이 있었습니다.  구독이 2번 필요한? 그런 케이스가 있을 경우 어떻게 해결해야 할 지 고민이 되네요.